src/main/cpp/src/decimal_utils.cu

```diff
@@ -675,22 +677,26 @@ struct dec128_multiplier {
 
     chunked256 product = multiply(a, b);
 
-    // Spark does some really odd things that I personally think are a bug
-    // https://issues.apache.org/jira/browse/SPARK-40129
-    // But to match Spark we need to first round the result to a precision of 38
-    // and this is specific to the value in the result of the multiply.
-    // Then we need to round the result to the final scale that we care about.
-    int dec_precision       = precision10(product);
-    int first_div_precision = dec_precision - 38;
-
-    int mult_scale = a_scale + b_scale;
-    if (first_div_precision > 0) {
-      auto const first_div_scale_divisor = pow_ten(first_div_precision).as_128_bits();
-      product                            = divide_and_round(product, first_div_scale_divisor);
-
-      // a_scale and b_scale are negative. first_div_precision is not
-      mult_scale = a_scale + b_scale + first_div_precision;
-    }
+    int dec_precision = precision10(product);
```

Comment:
`dec_precision` is used only once below. Please move it right above the line using it.

Reply:
Also please add `const` qualifier to all the variables.